int32 color
int32 point_x
int32 point_y
int32 rec_w
int32 rec_h
